src/npc_attack.cpp

```diff
@@ -486,23 +588,28 @@ npc_attack_rating npc_attack_throw::evaluate_tripoint(
     if( critter ) {
         att = source.attitude_to( *critter );
     }
-    const bool friendly_fire = att == Creature::Attitude::FRIENDLY &&
-                               !source.rules.has_flag( ally_rule::avoid_friendly_fire );
-    int attitude_mult = npc_attack_constants::attitude_multiplier.at( att );
-    if( friendly_fire ) {
-        // hitting a neutral creature isn't exactly desired, but it's a lot less than a friendly.
-        // if friendly fire is on, we don't care too much, though if an available hit doesn't damage them it would be better.
-        attitude_mult = npc_attack_constants::attitude_multiplier.at( Creature::Attitude::NEUTRAL );
+
+    if( att != Creature::Attitude::HOSTILE ) {
+        // No point in throwing stuff at neutral/friendly targets
+        return npc_attack_rating( cata::nullopt, location );
+    }
+
+    if( source.rules.has_flag( ally_rule::avoid_friendly_fire ) &&
+        !source.wont_hit_friend( location, thrown_item, false ) ) {
+        // Avoid friendy fire
+        return npc_attack_rating( cata::nullopt, location );
     }
 
     const float throw_mult = throw_cost( source, single_item ) * source.speed_rating() / 100.0f;
     const int damage = source.thrown_item_total_damage_raw( single_item );
     float dps = damage / throw_mult;
     const int distance_to_me = rl_dist( location, source.pos() );
-    const int distance_penalty = std::max( distance_to_me - 1,
-                                           ( source.closest_enemy_to_friendly_distance() - 1 ) * 2 );
+    const int distance_penalty = std::max( std::min( distance_to_me,
```

Comment:
use clamp instead of std::max( std::min())

Reply:
I couldn't get it to work in VS. It turns out std::clamp() is from C++17 and the project is set up for C++14. I'll make it more verbose so at least the min -> max thing doesn't impact readability.

Reply:
We have our own `clamp` in `cata_utility.h`.

Reply:
Thanks. Found it when I was reworking the tests and already updated the code to use clamp where applicable. I'll mark as resolved to avoid confusion.